integration-tests/rpc-get-protocol-constants.spec.ts

```diff
@@ -0,0 +1,438 @@
+import { TezosToolkit } from "@taquito/taquito";
+import { ConstantsResponse } from "@taquito/rpc";
+
+const rpcUrl = 'https://api.tez.ie/rpc/mainnet';
+const delphinetUrl = 'https://api.tez.ie/rpc/delphinet';
+const Tezos = new TezosToolkit(rpcUrl);
+const TezosDelphi = new TezosToolkit(delphinetUrl);
+
+describe('Fetch constants for all protocols on Mainnet', () => {
+    // tests
+    it('succesfully fails at fetching constants for level 0', async (done) => {
+        try {
+            await Tezos.rpc.getConstants({ block: "0" });
+            expect.assertions(1);
+        } catch (ex) {
+            expect(ex.message).toMatch('Http error response: (404) ');
+        }
+        done();
+    })
+
+    // TODO: Test that we aren't getting any extra properties. Currently, tests will pass even if we had extra properties in JSON response
```

Comment:
Could we implement it instead of keeping a TODO?

Instead of testing all properties separately, we could test if `constants` is equal to an object (if the object is missing one of the property, the test fail)
For example, the first test could be replaced by:
```
it('successfully fetches Proto1 constants at level 1', async (done) => {

            // Get constants for protocol
            const constants = await Tezos.rpc.getConstants({ block: "1" });
            expect(constants).toEqual({
                proof_of_work_nonce_size: 8,
                nonce_length: 32,
                max_revelations_per_block: 32,
                max_operation_data_length: 16384,
                preserved_cycles: 5,
                blocks_per_cycle: 4096,
                blocks_per_commitment: 32,
                blocks_per_roll_snapshot: 256,
                blocks_per_voting_period: 32768,
                time_between_blocks: [new BigNumber(60), new BigNumber(75)],
                endorsers_per_block: 32,
                hard_gas_limit_per_operation: new BigNumber(400000),
                hard_gas_limit_per_block:  new BigNumber(4000000),
                proof_of_work_threshold:  new BigNumber(70368744177663),
                tokens_per_roll:  new BigNumber(10000000000),
                michelson_maximum_type_size: 1000,
                seed_nonce_revelation_tip:  new BigNumber(125000),
                origination_burn: '257000',
                block_security_deposit:  new BigNumber(0),
                endorsement_security_deposit:  new BigNumber(0),
                block_reward:  new BigNumber(0),
                endorsement_reward:  new BigNumber(0),
                cost_per_byte:  new BigNumber(1000),
                hard_storage_limit_per_operation:  new BigNumber(60000),
            });

        done();
    })
```

We could make this change to the following tests.
Let me know what you think?

Reply:
I added a check on the number of items expected. I also cleaned up the code.